src/app/archive/archive_lib/processor.ml

```diff
@@ -567,6 +567,78 @@ module Block = struct
               (coinbase_id, block_id)
         in
         return block_id
+
+  let delete_if_older_than ?height ?num_blocks ?timestamp
+      (module Conn : CONNECTION) =
+    let open Deferred.Result.Let_syntax in
+    let%bind height =
+      match (height, num_blocks) with
+      | Some height, _ ->
+          return height
+      | None, Some num_blocks -> (
+          match%map
+            Conn.find_opt
+              (Caqti_request.find_opt Caqti_type.unit Caqti_type.int
+                 "SELECT MAX(height) FROM blocks")
+              ()
+          with
+          | Some max_block_height ->
+              max_block_height - num_blocks
+          | _ ->
+              0 )
+      | None, None ->
+          return 0
+    in
+    let timestamp = Option.value ~default:Int64.zero timestamp in
+    let%bind () =
```

Comment:
Can we stick all these deletes (and the select max) inside a single transaction? Without that, we may leave the db in a weird state if something breaks in the middle of these SQL commands.

Reply:
I think the `Conn.start`/`Conn.commit` in `archive.ml` give us this already, but I also don't know SQL that well.